### PR TITLE
Rename "metaData" to "metadata"

### PR DESCRIPTION
**Pull request by robinmacharg**
## Goal

Rename internal occurrences of "metaData" with "metedata" for cross-platform consistency and as a basis for subsequent work.  This change leaves alone any occurrences that are destined for the Bugsnag HTTP API since this has not changed.  For completeness: If API-related occurrences are changed at a later date they will have to account for both "metaData" and "metadata" validly existing in on-disk events.

## Design

Simple manual refactor, file by file, using repeated case-sensitive text search and avoiding API interacting occurrences.  One gotcha was the mention in the `module.modulemap` that caused compilation errors with erroneous root cause.

## Changeset

Multiple files affected (not all visible from an XCode full-text search):

```
M	OSX/module.modulemap
M	Source/Bugsnag.h
M	Source/Bugsnag.m
M	Source/BugsnagBreadcrumb.m
M	Source/BugsnagConfiguration.h
M	Source/BugsnagConfiguration.m
M	Source/BugsnagEvent.h
M	Source/BugsnagEvent.m
M	Source/BugsnagKeys.h
M	Source/BugsnagMetadata.h
M	Source/BugsnagMetadata.m
M	Source/BugsnagNotifier.h
M	Source/BugsnagNotifier.m
M	Tests/BugsnagEventTests.m
M	Tests/BugsnagSinkTests.m
M	iOS/BugsnagTests/BugsnagEventFromKSCrashReportTest.m
M	iOS/module.modulemap
M	tvOS/module.modulemap
```

## Tests

Existing tests were modified for consistency and all passed.

### Outstanding Questions

- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [X] Final review

- [X] The correct target branch has been selected (`master` for fixes, `next` for
  features)
- [ ] Consistency across platforms for structures or concepts added or modified
- [X] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: iOS/BugsnagTests/BugsnagKSCrashSysInfoParserTest.m [hunk 1]
```diff
@@ -16,14 +16,14 @@ @interface BugsnagKSCrashSysInfoParserTest : XCTestCase
...
 
-- (void)testEmptyDictSerialisation {
+- (void)XtestEmptyDictSerialisation {
     // ensures that an empty dictionary parameter returns a fallback dictionary populated with at least some information
     NSDictionary *device = BSGParseDevice(@{});
     [self validateDeviceDict:device];
 }
 
 
-- (void)testNilDictSerialisation {
```
Comment (kattrali) [inline iOS/BugsnagTests/BugsnagKSCrashSysInfoParserTest.m]: might have missed restoring these

**Comment by robinmacharg** [comment]
> > This change leaves external API-related occurrences as they are.
>
> What does this part mean? As in the HTTP public interface or library public interface?

Correct: HTTP.  Any occurrences of "metaData" - capital-D - that are sent to the main Bugsnag API have been left as-is.  I'll edit the commit message to clarify this.

File: iOS/BugsnagTests/BugsnagKSCrashSysInfoParserTest.m (hunk 1 above)
Comment (robinmacharg) [reply iOS/BugsnagTests/BugsnagKSCrashSysInfoParserTest.m]: These tests have a dependency on a fix in another PR.  I'll update/reenable once that change is merged.

**Comment by kattrali** [comment]
The changeset itself looks good, the tests are either [failing](https://travis-ci.com/bugsnag/bugsnag-cocoa/jobs/283793665#L682) and/or [warning](https://travis-ci.com/bugsnag/bugsnag-cocoa/jobs/283793664#L800) because the Xcode project files for tvOS and macOS (tvOS/Bugsnag.xcodeproj and OSX/Bugsnag.xcodeproj) need to be updated with new filenames.

**Pull request merged**